app/controllers/application_controller.rb

```diff
@@ -259,6 +259,10 @@ def pull_pinned_links_and_hot_questions
             .order('score DESC').limit(SiteSetting['HotQuestionsCount']).all
       end
     end
+
+    # eager loading revived collections' used relation to prevent N+1 queries
+    @pinned_links = @pinned_links.list_includes
+    @hot_questions = @hot_questions.list_includes
```

Comment:
Eager loading is done in this manner here to faciliatate eager loading of collections reviced from cache.